Public/Src/Cache/ContentStore/Distributed/NuCache/FlushableCache.cs

```diff
@@ -137,6 +153,10 @@ private void PerformFlush(OperationContext context)
                     actionBlock.CompletionAsync().Wait();
                 }
 
+                statistics.FlushingTime = stopwatch.Elapsed;
+                statistics.Persisted = _flushingCache.Count;
+                stopwatch.Restart();
+
                 _database.Counters[ContentLocationDatabaseCounters.NumberOfPersistedEntries].Add(_flushingCache.Count);

```

Comment:
I think we're mixing two concepts together here: on one hand we have counters already, but on the other hand we track everything separately once again.

I'm thinking, does it make sense to compute one thing based on another?

For instance, I had a similar challenge in ContentLcoationEventStore, and what I did is that I've created new CounterCollection instance and used it instead of using separate stats class.

Reply:
In this case, I think, we'll be using one concept instead of two.

---
In reply to: [315392700](https://github.com/microsoft/BuildXL/pull/756#discussion_r315392700) [](ancestors = 315392700)

Reply:
I'm not sure I understand: so you're saying I should create a new CounterCollection? Isn't that overkill for a few, transient counters like these?

If you instead mean that I should compute these values based on the counters we already have, this is not possible right now because those don't track the values I'm interested in.

I could certainly add the necessary ones, but I don't see how that helps to log these values at the end of the flush.